### PR TITLE
user-defined KV pairs

### PR DESCRIPTION
**Pull request by douzzer**

### CODE REVIEW

File: src/kv.c [hunk 1]
```diff
@@ -0,0 +1,908 @@
+/*
+ * kv.c
+ *
+ * Copyright (C) 2021 wolfSSL Inc.
```
Comment (LinuxJedi) [inline src/kv.c:4]: 2022 now :)

File: src/kv.c [hunk 2]
```diff
@@ -0,0 +1,908 @@
...
+    struct wolfsentry_context *wolfsentry,
+    const char *key,
+    int key_len,
+    const char *value,
+    int value_len /* without terminating null */,
+    int overwrite_p)
+{
+    wolfsentry_errcode_t ret;
+    struct wolfsentry_kv_pair_internal *kv;
+    if (value_len < 0)
```
Comment (LinuxJedi) [inline src/kv.c:739]: <= maybe?

Comment (douzzer) [reply src/kv.c:739]: `value_len == 0` is valid -- zero-length value.

File: src/kv.c (hunk 1 above)
Comment (douzzer) [reply src/kv.c:4]: good point!  will update in next PR, codebase-wide.

File: src/kv.c (hunk 2 above)
Comment (LinuxJedi) [reply src/kv.c:739]: Resolved in conversation

File: src/kv.c (hunk 1 above)
Comment (LinuxJedi) [reply src/kv.c:4]: To be fixed in subsequent PR.

**Review by LinuxJedi — APPROVED** [review]

**Review by LinuxJedi — APPROVED** [review]

**Pull request merged**